src/mqtt_pal.c

```diff
@@ -106,6 +106,83 @@ ssize_t mqtt_pal_recvall(mqtt_pal_socket_handle fd, void* buf, size_t bufsz, int
 
 #endif
 
+#elif defined(_MSC_VER)
+
+#ifdef MQTT_USE_BIO
+#include <openssl/bio.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+ssize_t mqtt_pal_sendall(mqtt_pal_socket_handle fd, const void* buf, size_t len, int flags) {
+    size_t sent = 0;
+    while(sent < len) {
+        int tmp = BIO_write(fd, (char*)buf + sent, len - sent);
```

Comment:
There are only tiny differences between the `__unix__` and `MSVC` cases.
For one, I am surprised that it actually compiles when you add a value to a `void*` pointer on gcc...

Reply:
The `(char *)` cast has precedence over the addition. So this line is not adding a value to a void pointer. That would be illegal.

Reply:
yes, hence why I added the `char*` cast.
The gcc version however does not have this cast, so I am not sure why that works.

Reply:
Oh yeah, you're right. Mine was meant to be cast to a char as well.